### DISCUSSION TITLE
Can I auto-hide waybar?

### DISCUSSION THREAD

**Discussion by blakemcbride** [Alexays/Waybar#3225] [Q&A]
Greetings,

I would like to auto-hide waybar the same way many other window managers do with their status panels.  Can I do that with waybar?  If so, how?

Thanks!

**Comment by blakemcbride** [comment]
I found something that is nearly perfect as follows.  It only shows the bar when I hit the cmd button.

sway config:

```
bar {
    swaybar_command waybar
    position top
    hidden_state hide
    mode hide
    modifier Mod4
}
```
waybar config:
```
    "ipc": true,
```
The only problem I have with it is the bar's opacity.  I can still see what's behind it.  I don't like that.  How can I make it non-see-through?

**Comment by ordy** [comment]
> How can I make it non-see-through?

You mean in general? 
If the bar is transparent, just change the background-color to a solid color code or set 1 to rgba in your styles.css file
```
window#waybar {
    background-color: rgba(10,10,10,1);
    /* or */
    background-color: #101010;
}
```

**Comment by blakemcbride** [comment]
The following is what I ended up using and works perfectly for me:

```
window#waybar {
    background-color: rgba(43, 48, 59, 1.0);
    border-bottom: 3px solid rgba(100, 114, 125, 1.0);
    color: #ffffff;
    transition-property: background-color;
    transition-duration: .5s;
}
```

**Comment by parisni** [comment]
there is also this way to toggle waybar:
```
bindsym $mod+b exec "pkill -USR1 waybar"
```

**Reply by AyushmanOfficial** [reply]
Thanks!

**Comment by nightfury05** [comment]
u can also do this for toggle in hyprland config
bind = $mainMod, L, exec, pkill waybar || waybar

**Reply by AyushmanOfficial** [reply]
why do that when you can just SIGUSR?

**Comment by TheMrYah** [comment]
Yup, this is a cleaner and intended way to do it: bind = $mainMod, U, exec, killall -SIGUSR1 waybar

You can also make script for detecting your mouse position or focus of window to toggle it automatically.
